### ISSUE TITLE
isoquant failed in between run of pacbio data.

### ISSUE THREAD

**Issue by unique379r** [ablab/IsoQuant#99]
hi 
I am trying to run a bam (mapped with pbmm2) with the following command but it crashed between the run.

##Used command
```
isoquant.py --reference reference.fa --genedb reference.gtf --complete_genedb --bam m64480e_230124_161455.demux.bc1001--bc1001.pbmm2.sort.bam --data_type transcripts --fl_data --splice_correction_strategy default_pacbio --model_construction_strategy fl_pacbio --sqanti_output --count_exons --prefix m64480e_230124_161455.demux.bc1001--bc1001 --output test_isoquant_out
```

##ERROR

```
2023-07-11 18:14:03,633 - CRITICAL - IsoQuant failed with the following error, please, submit this issue to https://github.com/ablab/IsoQuant/issuesconcurrent.futures.process._RemoteTraceback:
"""
Traceback (most recent call last):
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/lib/python3.8/concurrent/futures/process.py", line 239, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/lib/python3.8/concurrent/futures/process.py", line 198, in _process_chunk
    return [fn(*args) for args in chunk]
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/lib/python3.8/concurrent/futures/process.py", line 198, in <listcomp>
    return [fn(*args) for args in chunk]
  File "/stornext/snfs4/next-gen/scratch/rupesh/Apps/envs/isoquant/share/isoquant-3.3.0-0/src/dataset_processor.py", line 228, in construct_models_in_parallel
    sqanti_t2t_printer.add_read_info(a)
  File "/stornext/snfs4/next-gen/scratch/rupesh/Apps/envs/isoquant/share/isoquant-3.3.0-0/src/assignment_io.py", line 410, in add_read_info
    read_assignment.exon_count(), match.match_classification.name, gene_id, transcript_id,
UnboundLocalError: local variable 'match' referenced before assignment
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/bin/isoquant.py", line 698, in <module>
    main(sys.argv[1:])
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/bin/isoquant.py", line 692, in main
    run_pipeline(args)
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/bin/isoquant.py", line 645, in run_pipeline
    dataset_processor.process_all_samples(args.input_data)
  File "/stornext/snfs4/next-gen/scratch/rupesh/Apps/envs/isoquant/share/isoquant-3.3.0-0/src/dataset_processor.py", line 368, in process_all_samples
    self.process_sample(sample)
  File "/stornext/snfs4/next-gen/scratch/rupesh/Apps/envs/isoquant/share/isoquant-3.3.0-0/src/dataset_processor.py", line 406, in process_sample
    self.process_assigned_reads(sample, saves_file)
  File "/stornext/snfs4/next-gen/scratch/rupesh/Apps/envs/isoquant/share/isoquant-3.3.0-0/src/dataset_processor.py", line 539, in process_assigned_reads
    for read_stat_counter, tsc in results:
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/lib/python3.8/concurrent/futures/process.py", line 484, in _chain_from_iterable_of_lists
    for element in iterable:
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/lib/python3.8/concurrent/futures/_base.py", line 619, in result_iterator
    yield fs.pop().result()
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/stornext/snfs5/next-gen/scratch/rupesh/Apps/envs/isoquant/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
UnboundLocalError: local variable 'match' referenced before assignment
```

**Comment by andrewprzh** [comment]
Dear @unique379r 

This is a known bug in SQANTI-like file output (previously reported in https://github.com/ablab/IsoQuant/issues/96). [This commit](https://github.com/ablab/IsoQuant/commit/c98725d83f4fe2070a4fa01754b5f46f642966cc)  fixes it, it will be included in the next release.

Best
Andrey

**Comment by andrewprzh** [comment]
Should be fixed now in version 3.3.1.
